packages/cli-kit/assets/cli-ruby/lib/shopify_cli/theme/extension/syncer/extension_serve_job.rb

```diff
@@ -43,24 +43,8 @@ def perform!
 
             job_in_progress!
 
-            input = {
-              api_key: @project.app.api_key,
-              registration_id: @project.registration_id,
-              config: JSON.generate(@specification_handler.config(@ctx)),
-              extension_context: @specification_handler.extension_context(@ctx),
-            }
-            response = ShopifyCLI::PartnersAPI.query(@ctx, "extension_update_draft", **input).dig(*RESPONSE_FIELD)
-            user_errors = response.dig(USER_ERRORS_FIELD)
-
-            if user_errors
-              @ctx.puts(error_message(@project.title))
-              error_files = erroneous_files(user_errors)
-              print_items(error_files)
-            else
-              @ctx.puts(success_message(@project.title))
-              print_items({}.freeze)
-              ::Extension::Tasks::Converters::VersionConverter.from_hash(@ctx, response.dig(VERSION_FIELD))
-            end
+            @ctx.puts(success_message(@project.title))
+            print_items({}.freeze)
```

Comment:
This seems like too simple a replacement.

The challenge is that this code is now no longer responsible for extension draft updates, and hot reloading of the preview seems to happen somewhere else.

I believe `::Extension::Tasks::Converters::VersionConverter.from_hash(@ctx, response.dig(VERSION_FIELD))` was meant to perform some kind of validation. Of course, `erroneous_files(user_errors)` is supposed to give us some additional output information as well, but we won't actually have access to the Partners response anymore, so only the TS code knows what went wrong (and should print that out).

However, given that the new system runs its own validations, including theme check, and any errors will be printed to the console, is it necessary to replace this content? Should we hide the output entirely? @karreiro I suspect you'd have some ideas here.

Reply:
That's an excellent point! I believe it's worth logging the files to keep a similar DX and also because we have the `--notify` flag (so it might help with debugging that). Personally, I'd keep this in TypeScript side with `removed:`, `created:`, `updated:` badges, as TS knows the filesystem now. But, I'm ok with keeping this line as is as well :)